### PR TITLE
workflows: holding pen details update

### PR DESCRIPTION
**Pull request by jalavik**
Updates to the Holding Pen details interface to have UI elements to delete and restart objects.

### CODE REVIEW

File: invenio/modules/workflows/models.py [hunk 1]
```diff
@@ -574,7 +574,7 @@ def get_action(self):
     def get_action_message(self):
         """Retrieve the currently assigned widget, if any."""
         try:
-            return self.get_extra_data()["_message"]
+            return unicode(self.get_extra_data()["_message"])
```
Comment (jirikuncar) [inline invenio/modules/workflows/models.py]: @jalavik please use unicodifier or other "safe" solution that can be easily migrated to PY2/3 support. Thanks

File: invenio/modules/workflows/utils.py [hunk 2]
```diff
@@ -472,12 +474,12 @@ def get_func_info(func):
     varnames = func.func_code.co_freevars
     if closure:
         for index, arg in enumerate(closure):
-            parameters.append((str(varnames[index]), str(arg.cell_contents)))
+            parameters.append((unicode(varnames[index]), unicode(arg.cell_contents)))
     return {
```
Comment (jirikuncar) [inline invenio/modules/workflows/utils.py]: @jalavik `unicodifier` can help you here ... `return unicodifier({...})` and recursive magic should happen.

File: invenio/modules/workflows/utils.py [hunk 3]
```diff
@@ -490,3 +492,19 @@ def get_workflow_info(func_list):
...
+
+
+def alert_response_wrapper(func):
+    """Wrap given function with wrapper to return JSON for alerts."""
+    @wraps(func)
+    def inner(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except Exception as error:
+            from flask import current_app, jsonify
```
Comment (jirikuncar) [inline invenio/modules/workflows/utils.py]: the import can be global

File: invenio/modules/workflows/views/holdingpen.py [hunk 4]
```diff
@@ -257,51 +263,69 @@ def restart_record(objectid, start_point='continue_next'):
         Workflow.uuid == bwobject.id_workflow).first()
 
     start_delayed(workflow.name, [bwobject.get_data()])
-    return 'Record Restarted'
+    return jsonify({
+        "category": "success",
+        "message": unicode(_("Object restarted successfully."))
```
Comment (jirikuncar) [inline invenio/modules/workflows/views/holdingpen.py]: don't use `unicode` here

Comment (jalavik) [inline invenio/modules/workflows/views/holdingpen.py]: Without it, the jsonifier would have problems dealing with the special lazy string type returned from  `_` and cause an error. I guess I could use anyway the `unicodifier`.

Comment (jirikuncar) [inline invenio/modules/workflows/views/holdingpen.py]: what does happen if you wrap it to `dict` first?

File: invenio/modules/workflows/views/holdingpen.py [hunk 5]
```diff
@@ -16,6 +16,7 @@
 # You should have received a copy of the GNU General Public License
 # along with Invenio; if not, write to the Free Software Foundation, Inc.,
 # 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
```
Comment (jirikuncar) [inline invenio/modules/workflows/views/holdingpen.py]: you can add `from __future__ import unicode_literals` to avoid some hacks above

Comment (jalavik) [inline invenio/modules/workflows/views/holdingpen.py]: Done

File: invenio/ext/babel/__init__.py [hunk 6]
```diff
@@ -102,6 +102,21 @@ def set_translations():
...
+    """Add a custom JSONEncoder that handles lazy strings from Babel."""
+    from speaklater import _LazyString
+    from six import text_type
+
+    class JSONEncoder(app.json_encoder):
+        def default(self, o):
+            if isinstance(o, _LazyString):
+                return text_type(o)
+
+            return app.json_encoder.default(self, o)
```
Comment (jirikuncar) [inline invenio/ext/babel/__init__.py]: either `return super(JSONEncoder, self).default(o)` or we need to keep a reference to the original `app.json_encoder`.

File: invenio/modules/workflows/templates/workflows/utils.html [hunk 7]
```diff
@@ -17,22 +17,22 @@
 # 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
```
Comment (jirikuncar) [inline invenio/modules/workflows/templates/workflows/utils.html]: copyright year ;)

Comment (jalavik) [inline invenio/modules/workflows/templates/workflows/utils.html]: Thanks, done

File: invenio/modules/workflows/static/js/workflows/templates/alert.mustache [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+  WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with Invenio; if not, write to the Free Software Foundation, Inc.,
+  59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+}}
+<div class="alert alert-{{category}} alert-dismissible" role="alert">
+  <button type="button" class="close" data-dismiss="alert"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
```
Comment (jirikuncar) [inline invenio/modules/workflows/static/js/workflows/templates/alert.mustache]: can you split the line to make it more readable?

**Comment by jirikuncar** [comment]
@jalavik it looks good :shipit:

**Comment by jalavik** [comment]
Cool. Rebased and re-pushed.

**Comment by jirikuncar** [comment]
@jalavik have you seen this error?
```
======================================================================
ERROR: test_halt_in_branch (invenio.modules.workflows.testsuite.test_workflows.WorkflowTasksTestAPI)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/inveniosoftware/invenio/invenio/modules/workflows/testsuite/test_workflows.py", line 156, in test_halt_in_branch
    eng = start('branchtest', data, module_name="unit_tests")
  File "/home/travis/build/inveniosoftware/invenio/invenio/modules/workflows/api.py", line 99, in start
    return run_worker(workflow_name, data, **kwargs)
  File "/home/travis/build/inveniosoftware/invenio/invenio/modules/workflows/worker_engine.py", line 45, in run_worker
    run_workflow(wfe=engine, data=objects, **kwargs)
  File "/home/travis/build/inveniosoftware/invenio/invenio/modules/workflows/client.py", line 92, in run_workflow
    raise exception_triggered
WorkflowError: WorkflowError(Error: TypeError("can't pickle function objects",)
Traceback (most recent call last):
  File "/home/travis/build/inveniosoftware/invenio/invenio/modules/workflows/engine.py", line 432, in processing_factory
    self.run_callbacks(callbacks, objects, obj)
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/workflow/engine.py", line 414, in run_callbacks
    self.run_callbacks(f, objects, obj, indent + 1)
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/workflow/engine.py", line 422, in run_callbacks
    self.execute_callback(f, obj)
  File "/home/travis/build/inveniosoftware/invenio/invenio/modules/workflows/engine.py", line 537, in execute_callback
    obj.set_extra_data(obj.extra_data)
  File "/home/travis/build/inveniosoftware/invenio/invenio/modules/workflows/models.py", line 373, in set_extra_data
    self._extra_data = base64.b64encode(cPickle.dumps(value))
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/copy_reg.py", line 70, in _reduce_ex
    raise TypeError, "can't pickle %s objects" % base.__name__
TypeError: can't pickle function objects
, id_workflow: 055adf64-eeb9-11e4-9e6a-1fa20abf96a9, id_object: 0, payload: '[]')

----------------------------------------------------------------------
```

**Comment by jalavik** [comment]
@jirikuncar Should be ready now. Tested OK with demosite.

**Pull request merged**